gitbook-docs/README.md

```diff
@@ -1,31 +1,54 @@
-
-    * Constraint : Binary file representing arithmetization of the circuit.
-    * Assignment Table: Binary file pre-processed with public inputs & witness.
-
-    The constraint and assignment table generated above should be passed as in input to proof generator binary. This will output a binary proof file.
-3. Proof verification is not part of the zkLLVM tool-chain currently, can be done via:&#x20;
-   1. Offline : Tooling to support validation of off-chain proof will be added in future.
-   2. On-chain : This involves a more steps which requires serialisation of the circuit and deployed on blockchain clusters. This flow of generating smart contracts is handled by the [lorem-ipsum](https://github.com/NilFoundation/lorem-ipsum-cli) project. A high level flow is described in the guides for [circuit developer](manual/getting-started/circuit-generation.md) & [proof verifier](manual/getting-started/proof-verifier.md).
+## `=nil;` zkLLVM Project
+
+zkLLVM is a compiler from high-level programming languages into input for provable computation protocols.
+Developers can use zkLLVM to generate inputs for any arbitrary zero-knowledge proof system or protocol
+that accepts input data in the form of algebraic circuits.
+It can be used with the Placeholder proof system — a derivative of PLONK —
+or any other compatible arithmetization.
+
+zkLLVM is an extension to the LLVM toolchain, supporting any frontend that compiles to
+[LLVM intermediate representation](https://llvm.org/docs/LangRef.html#introduction).
+With zkLLVM, developers can write code in native languages instead of DSLs specific to other tools.
+
+zkLLVM toolchain has two main tools:
+
+1.  `clang` compiles the program into a circuit — an intermediate representation in bytecode.
+2.  `assigner` creates an assignment table using the compiled circuit and input data.
+
+### Zero-knowledge proof workflow with zkLLVM
+
+zkLLVM toolchain is a crucial part of the zero-knowledge proof verification workflow.
+Here's how it all works.
+
+1.  First, the companies who want to prove transactions develop their code in a compatible language, like C++.
+    They compile it with zkLLVM into a circuit — a provable bytecode representation of the application.
+    
+    {% hint style="info" %}
+    `=nil;` recommends using [Crypto3 library](https://docs.nil.foundation/crypto3/)
+    for developing most performant circuit components.
+    {% endhint %}
+
+1.  To prove a particular transaction, the proof requester
+    combines the circuit with the inputs of that transaction in the `assigner`
+    and gets two binary files:
+    
+    * Constraint, which represents the arithmetization of the circuit.
+    * Assignment table, which contains pre-processed public inputs.
+
+3.  Proof producers use these files to verify the proof.
```

Comment:
1. Proof requesters are the users who will verify the proof. 
2. Proof verification is a binary operation true or false after which you would write your path-dependent business logic.